app/models/claim/matching_attribute_finder.rb

```diff
@@ -0,0 +1,41 @@
+class Claim
+  class MatchingAttributeFinder
```

Comment:
I like the implementation of this class; before I started reading it I had no idea how I would implement it if it were me; and this is really nice and readable.